pkgs/by-name/xn/xnconvert/package.nix

```
@@ -0,0 +1,62 @@
+{
+  lib,
+  fetchurl,
+  appimageTools,
+  makeDesktopItem,
+  imagemagick,
+  runCommand,
+}:
+
+let
+  icon =
+    runCommand "xnconvert-icon.png"
+      {
+        nativeBuildInputs = [ imagemagick ];
+        src = fetchurl {
+          url = "https://www.xnview.com/img/app-xnconvert-512.webp";
+          hash = "sha256-le+rvthQndY3KbkPYuMGZDDcvdpvH9CIS2REP1vmDXg=";
+        };
+      }
+      ''
+        convert $src $out
+      '';
+  desktopItem = (
+    makeDesktopItem {
+      name = "xnconvert";
+      desktopName = "XnConvert";
+      exec = "xnconvert";
+      icon = "xnconvert";
+      comment = "A fast, powerful and free cross-platform batch image converter.";
+      categories = [ "Graphics" ];
+    }
+  );
+in
+appimageTools.wrapType2 rec {
+  pname = "xnconvert";
+  version = "1.105.0";
+
+  src = fetchurl {
+    url = "https://download.xnview.com/old_versions/XnConvert/XnConvert-${version}.glibc2.17-x86_64.AppImage";
+    sha256 = "sha256-eWQSUVxR3G3XbwBCht6LW3t3/N668jH4UqK5OnRY0ko=";
```

`sha256` is depreciated in favor of SRI hash, thanksfully it is already in the right format, so only the attribute to change:
```suggestion
    hash = "sha256-eWQSUVxR3G3XbwBCht6LW3t3/N668jH4UqK5OnRY0ko=";
```